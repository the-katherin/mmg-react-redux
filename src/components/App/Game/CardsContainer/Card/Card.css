.card {
  margin: 5px;
  background-color: lightyellow;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center center;
  border: 1px solid lightgrey;
  box-shadow: 0 0 20px rgba(0,0,0,0.5);
  height: calc(80vh/3);
  width: calc(90vw/6);
  user-select: none;
  position: relative;
  transition: all .8s linear;
  transform-style: preserve-3d;
}

.card:hover{
  cursor:pointer;
}

/* default values */
.card,
 .card__back, 
 .card__front {
  height: calc(80vh/5);
  width: calc(90vw/8);
  border-radius: 15%;
}


.card--size3 {
  height: calc(80vh/3);
  width: calc(90vw/4);
 
}

.card--size6  {
  height: calc(80vh/4);
  width: calc(90vw/5);
 
}

.card--size12 {
  height: calc(80vh/6);
  width: calc(90vw/7);
  
}


 .card__back, 
 .card__front  {
  position: absolute;
  backface-visibility: hidden;
}

.card__front {
  transform: rotateY(180deg);
}

.card--match {
  border: 4px solid yellow;
  visibility: hidden;
}

.card--selected {
  transform: rotateY(180deg);
}

.card--not-selected {
  transform: rotateY(0deg);
}








