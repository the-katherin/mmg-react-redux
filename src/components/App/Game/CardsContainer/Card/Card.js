import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { cardsBack } from '../../../../../assets/img';

export class Card extends Component {

    render() {

        const size = `card--size${this.props.size}`;

        return (
            <div className={`card ${this.props.flipped} ${size} `} onClick={this.props.handleClick} cardnumber={this.props.cardNumber} id={this.props.id}>
                <img className={`card__back ${size}`} src={cardsBack[this.props.cardsBack]} alt="card back" />
                <img className={`card__front ${size}`} src={this.props.src} alt="card front" />
            </div>
        );
    }
};

Card.propTypes = {
    handleClick: PropTypes.func.isRequired,
    cardNumber: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
};

export default Card;