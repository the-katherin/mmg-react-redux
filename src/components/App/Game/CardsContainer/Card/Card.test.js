import React from "react";
import { Card } from "./Card";

let wrapper;

describe("Card", () => {
    beforeEach(() => {
        wrapper = shallow(<Card
            handleClick={jest.fn()}
            cardNumber={1}
            id={2}

        />)
    });

    test("match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });


});
