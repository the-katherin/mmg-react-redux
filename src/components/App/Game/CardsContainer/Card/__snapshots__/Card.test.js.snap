// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card match snapshot 1`] = `
<div
  cardnumber={1}
  className="card undefined card--sizeundefined "
  id={2}
  onClick={[Function]}
>
  <img
    alt="card back"
    className="card__back card--sizeundefined"
  />
  <img
    alt="card front"
    className="card__front card--sizeundefined"
  />
</div>
`;
